src/Type.hs

```diff
@@ -25,18 +25,25 @@ newMetaPtr = MetaPtr <$> newUnique
 instance Show MetaPtr where
   show (MetaPtr i) = "(MetaPtr " ++ show (hashUnique i) ++ ")"
 
-data TyF t
+data TypeConstructor
   = TSyntax
   | TSignal
   | TString
-  | TFun t t
-  | TMacro t
+  | TFun
+  | TMacro
   | TType
-  | TDatatype Datatype [t]
+  | TDatatype Datatype
   | TSchemaVar Natural
   | TMetaVar MetaPtr
+  deriving (Eq, Show)
+makePrisms ''TypeConstructor
+
+data TyF t = TyF
```

Comment:
Why is this called TyF now? The name before was because it was the pattern functor in a typical fixed point construction, but here it feels like the more relevant concept to readers of the code is that it's an application of a type constructor to some arguments. It's certainly also a pattern functor - but it seems that we now have a less-generic name available than we had before.

Reply:
(Purple bike shed please!)